int8 gridX
int8 gridY
int16 height
int16 width
float32 theta
